pages/detention-locations.vue

```diff
@@ -0,0 +1,228 @@
+       queryDispatcher.query(sparqlQuery).then(result => {
+
+         for (let i = 0; i < result.results.bindings.length; i++) {
+           let item = result.results.bindings[i];
+
+           let wikiPage = {
+             id: item.item.value,
+             pageTitle: item.page_title.value,
+           }
+
+           this.wikiPages.push(wikiPage);
+         }
+
+       });
+     },
+     setMarkersIcons: function () {
+       let Filtering = new FilteringMethods(this.filterProperties, "sex");
+
+       for (let i = 0; i < this.originalMarkers.length; i++) {
+         let marker = this.originalMarkers[i];
+         [
+           marker.markerIcon,
+           marker.active
+         ] = Filtering.getMarkerStateIconDependant(marker);
+       }
+     },
+     hasLocalStorageExpired: function () {
+       let hours = 24; // Reset when storage is more than 24hours
+       let now = new Date().getTime();
+       let setupTime = localStorage.getItem('setupTime');
+
+       return setupTime === null || (now - setupTime > hours * 60 * 60 * 1000);
+     },
+     loadDataFromLocalStorage: function () {
+       let allFilters = JSON.parse(localStorage.getItem('allFilters'));
+       this.filterProperties.socialClass.filters = allFilters.socialClass;
+       this.filterProperties.occupation.filters = allFilters.occupation;
+     },
+     loadData: async function () {
+       this.loadWikiEntries();
+       const icons = this.icons;
+
+       try {
+         let response = await myFetch('/main.php?type=detention')
+         console.log('API Response:', response);
+         this.queryOutput = response
+       } catch (e) {
+         Swal.fire({
+           title: 'Server Error',
+           html: '<div>We are unable to connect to the server to pull in map info. Please refresh the page and try again. If this error persists, please contact <a href="mailto:ltw-apps-dev.ed.ac.uk">ltw-apps-dev.ed.ac.uk</a></div>',
+           footer: 'witches.is.ed.ac.uk',
+           confirmButtonText: 'Close',
+           type: 'error',
+           showCloseButton: true,
+         });
+
+         return
+       }
+
+       let getData = new APIDataHandler(
+         this.queryOutput, this.wikiPages,
+         icons, null
+       );
+       let filtersFound = null;
+
+       [
+         this.originalMarkers,
+         filtersFound
+       ] = getData.loadAccussed('detention', this.filtersToFind);
+       this.filterProperties.socialClass.filters = filtersFound.socialClass;
+       this.filterProperties.occupation.filters = filtersFound.occupation;
+
+       this.setMarkersIcons();
+       this.loading = false;
+     }
+   },
+   mounted: function () {
+     this.loadData();
+   }
+ };
+</script>
+
+<style>
+ .cluster-img {
+   float: left;
+   width: 72px;
+   height: 55px;
+ }
+
+ .zoomed-in-img {
+   float: left;
+   width: 25px;
+   height: 38px;
+ }
+
+ .icon-shadow{
+   position: absolute;
+   top: 15px !important;
```

Comment:
Is the important definitely needed here and in the instance a few lines below? It always feels like a code smell when we need to use important in the CSS. I also might be wrong but I thought we could now scope CSS in vue.

Reply:
have removed, these styles aren't actually used in this page so unsure why they are there, should i delete?

Reply:
yes, if we don't need the css please delete it. Thanks Ruby

Reply:
done

Reply:
thank you